custom/enikshay/management/commands/run_diagnosis_test_migration.py

```diff
@@ -0,0 +1,108 @@
+            'datamigration_diagnosis_test_information',
+        ]
+
+        print("Starting {} migration on {} at {}".format(
+            "real" if commit else "fake", domain, datetime.datetime.utcnow()
+        ))
+
+        with open(log_path, "w") as log_file:
+            writer = csv.writer(log_file)
+            writer.writerow(headers)
+
+            for episode_case_id in accessor.get_case_ids_in_domain(type='episode'):
+                print('Looking at {}'.format(episode_case_id))
+                episode_case = accessor.get_case(episode_case_id)
+                case_properties = episode_case.dynamic_case_properties()
+
+                if self.should_migrate_case(case_properties):
+                    test = self.get_relevant_test_case(domain, episode_case)
+
+                    if test is not None:
+                        update = self.get_updates(test)
+                        print('Updating {}...'.format(episode_case_id))
+                        writer.writerow([episode_case_id] + [update[key] for key in headers[1:]])
+
+                        if commit:
+                            factory.update_case(case_id=episode_case_id, update=update)
+                    else:
+                        print('No relevant test found for episode {}'.format(episode_case_id))
+                else:
+                    print('Do not migrate {}'.format(episode_case_id))
+
+        print('Migration complete at {}'.format(datetime.datetime.utcnow()))
+
+    @staticmethod
+    def should_migrate_case(case_properties):
+        return (
+            case_properties.get('datamigration_diagnosis_test_information') != 'yes'
+            and case_properties.get('episode_type') == 'confirmed_tb'
+            and case_properties.get('treatment_initiated') == 'yes_phi'
+            and case_properties.get('diagnosis_test_result_date', '') == ''
+        )
+
+    @staticmethod
+    def get_relevant_test_case(domain, episode_case):
+        try:
+            occurrence_case = get_occurrence_case_from_episode(domain, episode_case.case_id)
+        except ENikshayCaseNotFound:
+            return None
+
+        indexed_cases = CaseAccessors(domain).get_reverse_indexed_cases([occurrence_case.case_id])
+        test_cases = [
+            case for case in indexed_cases
+            if case.type == 'test'
+            and case.get_case_property('rft_general') == 'diagnosis_dstb'
+            and case.get_case_property('result') == 'tb_detected'
+        ]
+        if test_cases:
+            return sorted(test_cases, key=lambda c: c.get_case_property('date_reported'))[-1]
+        else:
+            return None
+
+    @staticmethod
+    def get_updates(test):
+        test_case_properties = test.dynamic_case_properties()
+        return {
+            'diagnosis_test_result_date': test_case_properties.get('date_tested', ''),
+            'diagnosis_lab_facility_name': test_case_properties.get('testing_facility_name', ''),
```

Comment:
@nickpell 

It looks like a bunch of test cases don't have a testing_facility_name.  Can you change the migration to do something like:
`test.case_properties.get('testing_facility_name') or test.case_properties.get('testing_facility_saved_name') or test.case_properties.get('test_facility_referred_to')`

Reply:
(note) this is being handled here:
https://github.com/dimagi/commcare-hq/pull/17739#pullrequestreview-61254331

Reply:
Yup, no need to make this change anymore..